docs/how_to_dev_frontend_en.md

```diff
@@ -0,0 +1,102 @@
+# VisualDL Web app Develop
+
+VisualDL has two three components.
+1. The Python/C++ SDK that logs the data during training.
+1. The single page client side app that visualized training data.
+1. The server(powered on Flask) that parses the saved data and delivers to the client side app for display.
+
+The server provided the needed data and pass to the client side to draw scalars or histograms. The users can visualize the training progress and make a better network architecture evaluation.
+
+This article will go over the basic web-app architecture and development steps.
+
+## npm package
+The VisualDL uses npm to manage the JavaScript files. npm is the package manager for JavaScript and is a great tool to manage multiple dependencies. To confirm that you have npm installed. Please run this command
+``` bash
+npm -v
+```
+If your environment does not have npm already, please [download npm](https://www.npmjs.com/get-npm).
+
+The basic packaging information is stored in `package.json`. It specifies build scripts and dependencies for each environment.
+npm reads `package.json` to build different package according to the packaging script.
+If you are not familiar with npm, please refer to the [npm documentation](https://docs.npmjs.com/) for more detail.
+
+Here is a list of the dependencies used in the VisualDL.
+```
+"axios": "^0.16.1",
+"csshint": "^0.3.3",
+"d3-format": "^1.2.1",
+"echarts": "^3.8.5",
+"file-saver": "^1.3.3",
+"htmlcs": "^0.4.1",
+"lesslint": "^1.0.2",
+"lodash": "^4.17.4",
+"normalize.css": "^6.0.0",
+"qs": "^6.5.1",
+"san": "3.2.3",
+"san-mui": "^1.0.4",
+"san-router": "^1.1.1",
+"san-store": "^1.0.1",
+"san-update": "^2.1.0",
+"xlsx": "^0.11.3"
+```
+
+## Development Flow
+The VisualDL is an open source project.
+Everyone is welcome to contribute to the project.
+To get start the process, please refer to [Contribute Code](contribute_code_en.md) to set up the environment.
+
+####Run The Client Side App In Development
```

Comment:
Lets remove this step and add the new steps:

Prerequisite: Python 2.7, pip, npm, cmake

git clone git@github.com:PaddlePaddle/VisualDL.git
cd VisualDL
pip install -r requirements.txt
./scripts/setup_dev_env.sh
./scripts/start_dev_server.sh --logdir=<LOG_DIR>